### ISSUE TITLE
Support default value syntax for ctx/MDC reference

### ISSUE THREAD

**Issue by hijackit** [tkowalcz/tjahzi#54]
Log4j2 allows the definition of a "fallback" value when performing property substitution:
https://logging.apache.org/log4j/2.x/manual/configuration.html#DefaultProperties
And the syntax is quite simple:
`${lookupName:key:-defaultValue}`

When it comes to MDC properties, which are computed at runtime, it makes sense to allow a default fallback property, but tjhazi does not allow this yet.
My use case is always the "tenant" property: sometimes the tenant may not be present such for those log messages related to the system (e.g. application startup, periodic tasks, etc..).

Long story short, I would like to be able to write something like this:
`<Label name="tenant" value="${ctx:tenant:-system}"/>`

**Comment by tkowalcz** [comment]
Hi,

I use log4j2 internal classes (`Interpolator` and `StrSubstitutor`) to implement ctx/MDC variable interpolation so this should already work. I have added test for that and "it works for me" -https://github.com/tkowalcz/tjahzi/commit/2e9d4adc6711e4fdaa9815b827e82cff0d2d18f4. 

I also tested with your exact label example: `<Label name="tenant" value="${ctx:tenant:-system}"/>` .

I am open to further investigate this issue but will need some more info how you tested.

**Comment by hijackit** [comment]
Thanks for the reply. I still have to check what is wrong with my code

**Comment by tkowalcz** [comment]
I hope you were able to find the issue. I will close this ticket but don't hesitate to reopen it if needed.

**Comment by hijackit** [comment]
Thanks, I successfully obtained what I needed but I had to use the following syntax with two leading dollars:
`<Label name="tenant" value="$${ctx:tenant:-system}"/>`

It's not clear to me if this is how it is supposed to work, but I think that is a matter of how log4j2 works.

**Comment by tkowalcz** [comment]
I'm glad you got it finally working, but that's very strange.
